tests/keras/test_keras_model_export.py

```diff
@@ -506,3 +507,38 @@ def test_sagemaker_docker_model_scoring_with_default_conda_env(model, model_path
     deployed_model_preds = pd.DataFrame(json.loads(scoring_response.content))
 
     np.testing.assert_array_almost_equal(deployed_model_preds.values, predicted, decimal=4)
+
+
+def test_save_model_with_tf_save_format(model_path):
+    """Ensures that Keras models can be saved with SavedModel format.
+
+    Using SavedModel format (save_format="tf") requires that the file extension
+    is _not_ "h5".
+    """
+    keras_model = mock.Mock(spec=tf.keras.Model)
+    mlflow.keras.save_model(keras_model=keras_model, path=model_path, save_format="tf")
+    _, args, kwargs = keras_model.save.mock_calls[0]
+    # Ensure that save_format propagated through
+    assert kwargs["save_format"] == "tf"
+    # Ensure that the saved model does not have h5 extension
+    assert not args[0].endswith(".h5")
+
+
+@pytest.mark.large
+def test_save_and_load_model_with_tf_save_format(tf_keras_model, model_path):
+    """Ensures that keras models saved with save_format="tf" can be loaded."""
+    mlflow.keras.save_model(keras_model=tf_keras_model, path=model_path, save_format="tf")
+    model_conf_path = os.path.join(model_path, "MLmodel")
+    model_conf = Model.load(model_conf_path)
+    flavor_conf = model_conf.flavors.get(mlflow.keras.FLAVOR_NAME, None)
+    assert flavor_conf is not None
+    assert flavor_conf.get("save_format") == "tf"
+    assert not os.path.exists(
+        os.path.join(model_path, "data", "model.h5")
+    ), "TF model was saved with HDF5 format; expected SavedModel"
```

Comment:
These assert error messages are actually nice because they clarify what is being asserted :)